src/magnify/apps/core/serializers/rooms.py

```diff
@@ -44,18 +61,42 @@ class RoomUserAccessSerializer(RoomAccessSerializerMixin, serializers.ModelSeria
 
     class Meta:
         model = models.RoomUserAccess
-        fields = ["id", "user", "room", "is_administrator"]
+        fields = ["id", "user", "room", "role"]
         read_only_fields = ["id"]
 
+    def update(self, instance, validated_data):
+        """Make user and room field readonly but only on update."""
+        validated_data.pop("room", None)
+        validated_data.pop("user", None)
+        return super().update(instance, validated_data)
```

Comment:
as it is common with RoomGroupAccessSerializer maybe it could inherit from a mixin

Reply:
it's common but not really because it is with different fields... so that's the most straightforward?